notes.go

```diff
@@ -1,41 +1,147 @@
 package main
 
 import (
-	"notes-service/grpc/notespb"
 	"context"
+	"fmt"
+	"notes-service/grpc/notespb"
 
+	"go.mongodb.org/mongo-driver/bson/primitive"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/emptypb"
-)
 
+	"go.mongodb.org/mongo-driver/bson"
+)
 
 type notesService struct {
 	notespb.UnimplementedNotesServiceServer
 }
 
 var _ notespb.NotesServiceServer = &notesService{}
 
-func (srv *notesService) CreateNote(ctx context.Context, in *notespb.Note) 
-(*emptypb.Empty, error)
-{
-	return nil, status.Errorf(codes.Unimplemented, "not implemented")
+func (srv *notesService) GetNote(ctx context.Context, in *notespb.GetNoteRequest) (*notespb.Note, error) {
+
+	objID, err := primitive.ObjectIDFromHex(in.Id)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Error : during get note 1")
+	}
+
+	var note *notespb.Note
+
+	err = NotesCollection.FindOne(context.TODO(), bson.M{"_id": objID}).Decode(&note)
+
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+
+	return note, status.Errorf(codes.OK, "Success : Note found")
+}
+
+func (srv *notesService) CreateNote(ctx context.Context, in *notespb.Note) (*emptypb.Empty, error) {
+
+	_, err := NotesCollection.InsertOne(
+		ctx,
+		bson.D{
+			{"authorId", in.AuthorId},
+			{"title", in.Title},
+			{"blocks", in.Blocks},
+		})
```

Comment:
Big nitpicking but in my opinion you shouldn't make this statement span that many lines. Compact it.
```suggestion
	_, err := NotesCollection.InsertOne(ctx, bson.D{
			{"authorId", in.AuthorId},
			{"title", in.Title},
			{"blocks", in.Blocks},
		})
```